src/gebt_poc/solver.cpp

```diff
@@ -0,0 +1,236 @@
+    const Kokkos::View<double*> gen_coords_derivatives,
+    const Kokkos::View<double**> sectional_stiffness
+) {
+    // Calculate first part of the elastic forces i.e. F^C vector
+    auto sectional_strain_next = Kokkos::View<double*>("sectional_strain_next", 6);
+    Kokkos::deep_copy(sectional_strain_next, sectional_strain);
+    auto R_x0 = gen_alpha_solver::multiply_matrix_with_vector(
+        rotation.GetRotationMatrix(),
+        gen_alpha_solver::create_vector(
+            {position_vector_derivatives(0), position_vector_derivatives(1),
+             position_vector_derivatives(2)}
+        )
+    );
+    Kokkos::parallel_for(
+        3, KOKKOS_LAMBDA(const size_t k) { sectional_strain_next(k) -= R_x0(k); }
+    );
+
+    auto elastic_force_fc =
+        gen_alpha_solver::multiply_matrix_with_vector(sectional_stiffness, sectional_strain_next);
+
+    // Calculate second part of the elastic forces i.e. F^D vector
+    auto x0_prime_tilde =
+        gen_alpha_solver::create_cross_product_matrix(gen_alpha_solver::create_vector(
+            {position_vector_derivatives(0), position_vector_derivatives(1),
+             position_vector_derivatives(2)}
+        ));
+    auto u_prime_tilde =
+        gen_alpha_solver::create_cross_product_matrix(gen_alpha_solver::create_vector(
+            {gen_coords_derivatives(0), gen_coords_derivatives(1), gen_coords_derivatives(2)}
+        ));
+    auto fd_values = gen_alpha_solver::transpose_matrix(
+        gen_alpha_solver::add_matrix_with_matrix(x0_prime_tilde, u_prime_tilde)
+    );
+
+    auto elastic_force_fd = Kokkos::View<double*>("elastic_force_fd", 6);
+    Kokkos::deep_copy(elastic_force_fd, 0.);
+    Kokkos::parallel_for(
+        Kokkos::MDRangePolicy<Kokkos::DefaultExecutionSpace, Kokkos::Rank<2>>(
+            {0, 0}, {kNumberOfVectorComponents, kNumberOfVectorComponents}
+        ),
+        KOKKOS_LAMBDA(const size_t i, const size_t j) {
+            elastic_force_fd(i + 3) += fd_values(i, j) * elastic_force_fc(j);
+        }
+    );
+
+    auto elastic_forces = Kokkos::View<double*>("elastic_forces", 12);
+    Kokkos::parallel_for(
+        6,
+        KOKKOS_LAMBDA(const size_t k) {
+            elastic_forces(k) = elastic_force_fc(k);
+            elastic_forces(k + 6) = elastic_force_fd(k);
+        }
+    );
+    return elastic_forces;
+}
+
+Kokkos::View<double*> CalculateStaticResidual(
+    const Kokkos::View<double*> position_vectors, const Kokkos::View<double*> gen_coords,
+    const StiffnessMatrix& stiffness, const Quadrature& quadrature
+) {
+    const auto n_nodes = gen_coords.extent(0) / kNumberOfLieAlgebraComponents;
+    const auto order = n_nodes - 1;
+    const auto n_quad_pts = quadrature.GetNumberOfQuadraturePoints();
+
+    auto residual = Kokkos::View<double* [6]>("static_residual");
+    Kokkos::deep_copy(residual, 0.);
+    for (size_t i = 0; i < n_nodes; ++i) {
+        for (size_t j = 0; j < n_quad_pts; ++j) {
```

Comment:
I think that we want this function to take in a field data and mesh and have the outer loop be over elements.  For example, we can do:

```
Kokkos::parallel_for(field_data.GetNumberOfElements(), KOKKOS_LAMBDA(int i) {
  //when we need a node loop
  auto nodes = mesh.GetNodesForElement(i);
  for(int j = 0; j < nodes.extent(0); ++j) { 
    auto data =  field_data.GetNodalData<FieldName>(j);
    //do work
  }

  //when we need a quadrature point loop
  auto data = field_data.GetElementData<FieldName>(i);
  for(int j = 0; j < quadrature.GetNumberOfQuadraturePoints(); ++j) {
    //do work, perhaps even getting nodes for the element and looping them in an inner loop
  }
  
});
```

Residual should be a `Kokkos::View<double*>` instead of a `Kokkos::View<double*[6}>` to facilitate use with linear algebra functions.  
 
Also, let's not create the residual in this function, but have it be an output argument.  We don't want to reallocate it every time we call "CalculateStaticResidual."  Dynamic allocation is generally quite slow, and this vector will generally be quite big.

Reply:
@ddement Above makes sense to me, let's make these changes once the field data, mesh, and gen-alpha algorithms are solidified and merged into main. I already made the change to `Kokkos::View<double*>`, rest shouldn't be too hard to tweak to the needs of the solver workflow and DS. Sounds good?